### ISSUE TITLE
`patch list` is not so usable when tons of patches are available

### ISSUE THREAD

**Issue by andy-shev** [getpatchwork/git-pw#27]
Here is an issue with `git pw patch list`.
It can't show properly the limit of commits. `--limit=0` does not affect anything (I'm expecting no limit, or server defined limit), `--limit=1000` shows 1000 patches, however, I would rather need only few patches by given submitter (okay, this wouldn't work with API 1.0), or for given date (or period in time).
So, `git pw patch list --date=today` or alike is what being missed.

**Comment by stephenfin** [comment]
Good ideas. I guess there are a two things to do here.

- Special case `--limit=0` to mean "use server default"
- Add filtering by date, presumably using `--since` and `--before` arguments, and add a special casing for `today` (I like this idea).

Make sense?

Filtering by submitter should work with API 1.0 but requires a second call to the API. What doesn't work is filtering by multiple submitters.

**Comment by andy-shev** [comment]
Regarding ideas.
1. Yes.
2. Something like that and perhaps `--since=<ISO 8601>` and so on.

**Comment by andy-shev** [comment]
Regarding an issue.
`--limit=1000 --submitter 'First Last'` (or basically whatever more than page limit) doesn't show new patches at all, some strange list of them (like last page?) on API 1.0. Tested with *linux-gpio* project @ patchwork.ozlabs.org.

**Comment by andy-shev** [comment]
Done.
https://github.com/getpatchwork/patchwork/issues/229

**Comment by stephenfin** [comment]
> Done.
> [getpatchwork/patchwork#229](https://github.com/getpatchwork/patchwork/issues/229)

I checked this out. This is a known issue with Patchwork API 1.0 that was resolved in API 1.1 (included in Patchwork 2.1). Unfortunately the ozlabs instance hasn't been updated yet. However, you should have seen a warning like the below.

```
$ git pw patch list --submitter 'Wolfram Sang'
More than one match for found for `--submitter=Wolfram Sang`. Filtering by multiple submitters is not supported with API version 1.0. If the server supports it, use version 1.1 instead. Refer to https://git.io/vN3vX for more information.
+--------+---------------+--------------------------------------------------------------------------+----------------------------------+---------+------------+------------+
|     ID | Date          | Name                                                                     | Submitter                        | State   | Archived   | Delegate   |
|--------+---------------+--------------------------------------------------------------------------+----------------------------------+---------+------------+------------|
| 857404 | 10 months ago | [2/2] Documentation: gpio: correct return value of gpiod_get_direction   | Wolfram Sang (wsa@the-dreams.de) | new     | no         |            |
| 857403 | 10 months ago | [1/2] gpio: correct docs about return value of gpiod_get_direction       | Wolfram Sang (wsa@the-dreams.de) | new     | no         |            |
...
```

Did you not? If so, what version are you using?

**Comment by stephenfin** [comment]
Resolved in #67
